packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -198,7 +198,8 @@ contract TaikoL1 is EssentialContract, ITaikoL1, TaikoEvents, TaikoErrors {
             // Taiko blocks proposed per Ethereum block is smaller than 1.
             blockMaxGasLimit: 240_000_000,
             livenessBond: 250e18, // 250 Taiko token
-            blockSyncThreshold: 16
+            blockSyncThreshold: 16,
+            onlyEOACanUseCalldataForDA: true
```

Comment:
We using this flag for not breaking tests ?
Bc. I think this flag is not really needed because we need EOAs now too. (even tho different check what we using now:  `msg.sender != tx.origin`)

Reply:
> We using this flag for not breaking tests ?
Yes

Reply:
> Bc. I think this flag is not really needed because we need EOAs now too. (even tho different check what we using now: msg.sender != tx.origin)

Didn't get your point, please elaborate a bit.

Reply:
I meant that basically even now (pre-Petra) we dont want contracts to be the proposers, because this way, retrieving calldata back is hard (from client perspective).
So in testnet/mainnet environment this flag will always has to be true.

Reply:
If it is only needed to distinguish between test suite (foundry test) vs. runnig testnet/mainnet environment, maybe we can reuse the `livenessBond == 1 TKO` -> which is 1 in our `TaikoL1LibProvingWithTiers.t.sol `and `TaikoL1.t.sol` and remove this flag completely. What do you think ?

Reply:
Using `livenessBond == 1` is like a hidden rule, using a dedicated flag makes it more explicit.

Besides, we may want to skip EOA check and set livenessBond to a value other than 1, then the trick will stop working.

Reply:
Fixed it differently, please see the new commit.